internal/restorer/progressprinter/progressprinter.go

```diff
@@ -0,0 +1,114 @@
+package progressprinter
+
+import (
+	"fmt"
+	"sync"
+	"os"
+	"io"
+	"time"
+)
+
+var progresswriter io.Writer = os.Stdout
+
+// \r = carriage return (cursor to start of line);  \033[K = line deletion (delete from cursor until end of line)
+var lineResetter string = "\r\033[K"
+
+type progressInfoEntry struct {
+	name         string
+	bytesWritten int64
+	bytesTotal   int64
+}
+
+type RestoreProgressPrinter struct {
+	progressInfoMap  map[string]progressInfoEntry
+	filesFinished    int64
+	filesTotal       int64
+	allBytesWritten  int64
+	allBytesTotal    int64
+	started          time.Time
+	sync.Mutex
```

Comment:
It's customary to put the mutex before the fields that it protects, i.e., at the start of the struct:

```suggestion
	sync.Mutex
	progressInfoMap  map[string]progressInfoEntry
	filesFinished    int64
	filesTotal       int64
	allBytesWritten  int64
	allBytesTotal    int64
	started          time.Time
```